daemon/graphdriver/overlay2/overlay.go

```diff
@@ -174,6 +175,14 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		}
 	}
 
+	// If kernel is older than 4.8, check whether opaque copy up bug has been patched
+	if kernel.CompareKernelVersion(*v, kernel.VersionInfo{Kernel: 4, Major: 8, Minor: 0}) < 0 {
```

Comment:
We could do this like https://github.com/docker/docker/blob/master/daemon/graphdriver/aufs/aufs.go#L643 with `sync.Once` and lazy loading. Kernel check could probably be skipped then.